api/database/database.go

```diff
@@ -0,0 +1,53 @@
+package database
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/golang-migrate/migrate/v4"
+	"github.com/jinzhu/gorm"
+
+	// required for gomigrate
+	"github.com/golang-migrate/migrate/v4/database/postgres"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+
+	// enable postgres dialect
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+
+	"github.com/gojek/mlp/api/config"
+)
+
+// InitDB initialises a database connection as well as runs the migration scripts.
+// It is important to close the database after using it by calling defer db.Close()
+func InitDB(dbCfg *config.DatabaseConfig) (*gorm.DB, error) {
+	db, err := gorm.Open(
+		"postgres",
+		fmt.Sprintf("host=%s port=%d user=%s dbname=%s password=%s sslmode=disable",
+			dbCfg.Host,
+			dbCfg.Port,
+			dbCfg.User,
+			dbCfg.Database,
+			dbCfg.Password))
+	if err != nil {
+		panic(err)
```

Comment:
You don't need to panic here. Error handling should be done in `main`

```go
if err != nil {
   return nil, err
}
```

Reply:
fixed, thanks 👍